src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -219,7 +220,7 @@ private async void DiagnosticService_DiagnosticsUpdated(object sender, Diagnosti
             }
         }
 
-        private async Task PublishDiagnosticsAsync(Document document)
+        protected virtual async Task PublishDiagnosticsAsync(Document document)
         {
             var diagnostics = await GetDiagnosticsAsync(document, CancellationToken.None).ConfigureAwait(false);
             var publishDiagnosticsParams = new PublishDiagnosticParams { Diagnostics = diagnostics, Uri = document.GetURI() };
```

Comment:
This `Uri = document.GetURI()` piece doesn't return the re-mapped document location.  However it does have the correct line location now 😄 

![image](https://user-images.githubusercontent.com/2008729/76656609-3d06a200-652d-11ea-926e-c3ca08e08eab.png)